tests/test_utils.py

```diff
@@ -118,3 +119,112 @@ def test_endpoint_config():
+         db: rasa
+         user: user
+         password: pass
+         collection: conversations
+    """
+    r = utils.read_yaml_string(config_without_env_var)
+    assert r['tracker_store']['mongo']['user'] == 'user' and \
+        r['tracker_store']['mongo']['password'] == 'pass'
+
+
+def test_read_yaml_string_with_env_var():
+    os.environ['USER_NAME'] = 'user'
+    os.environ['PASS'] = 'pass'
+
+    config_with_env_var = """
+    tracker_store:
+      mongo:
+         host: http://localhost:27017
+         db: rasa
+         user: ${USER_NAME}
+         password: ${PASS}
+         collection: conversations
+    """
+    r = utils.read_yaml_string(config_with_env_var)
+    assert r['tracker_store']['mongo']['user'] == 'user' and \
+        r['tracker_store']['mongo']['password'] == 'pass'
+
+
+def test_read_yaml_string_with_env_var_prefix():
+    os.environ['USER_NAME'] = 'user'
+    os.environ['PASS'] = 'pass'
+
+    config_with_env_var = """
+    tracker_store:
+      mongo:
+         host: http://localhost:27017
+         db: rasa
+         user: db_${USER_NAME}
+         password: db_${PASS}
+         collection: conversations
+    """
+    r = utils.read_yaml_string(config_with_env_var)
+    assert r['tracker_store']['mongo']['user'] == 'db_user' and \
+        r['tracker_store']['mongo']['password'] == 'db_pass'
+
+
+def test_read_yaml_string_with_env_var_postfix():
+    os.environ['USER_NAME'] = 'user'
+    os.environ['PASS'] = 'pass'
+
+    config_with_env_var = """
+    tracker_store:
+      mongo:
+         host: http://localhost:27017
+         db: rasa
+         user: ${USER_NAME}_admin
+         password: ${PASS}_admin
+         collection: conversations
+    """
+    r = utils.read_yaml_string(config_with_env_var)
+    assert r['tracker_store']['mongo']['user'] == 'user_admin' and \
+        r['tracker_store']['mongo']['password'] == 'pass_admin'
+
+
+def test_read_yaml_string_with_env_var_infix():
+    os.environ['USER_NAME'] = 'user'
+    os.environ['PASS'] = 'pass'
+
+    config_with_env_var = """
+    tracker_store:
+      mongo:
+         host: http://localhost:27017
+         db: rasa
+         user: db_${USER_NAME}_admin
+         password: db_${PASS}_admin
+         collection: conversations
+    """
+    r = utils.read_yaml_string(config_with_env_var)
+    assert r['tracker_store']['mongo']['user'] == 'db_user_admin' and \
+        r['tracker_store']['mongo']['password'] == 'db_pass_admin'
+
+
+def test_read_yaml_string_with_env_var_not_exist():
+    os.environ['USER_NAME'] = 'user'
+    os.environ['PASS'] = 'pass'
+
+    config_with_env_var = """
+    tracker_store:
+      mongo:
+         host: http://localhost:27017
+         db: rasa
+         user: ${USER_NAME}
+         password: ${PASSWORD}
+         collection: conversations
+    """
+    try:
+        r = utils.read_yaml_string(config_with_env_var)
+    except Exception as e:
+        assert isinstance(e, KeyError) and e.args[0] == 'PASSWORD'
```

Comment:
please use the pytest method to check for exceptions: https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
@leon0707